core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -74,6 +74,12 @@
  * @see jenkins.model.Jenkins#proxy
  */
 public final class ProxyConfiguration extends AbstractDescribableImpl<ProxyConfiguration> implements Saveable, Serializable {
+    /**
+     * Holds a default TCP connect timeout set on all connections returned from this class,
+     * note this is value is in milliseconds, it's passed directly to {@link URLConnection#setConnectTimeout(int)}
+     */
+    private static final int DEFAULT_CONNECT_TIMEOUT_MILLIS = Integer.getInteger("jenkins.proxy.default_connect_timeout_millis", 20 * 1000);
```

Comment:
Naming inconsistent with all other System properties we have: https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties

It's camelCase except in the case of a STATIC_FINAL_FIELD_NAME.